sqlmesh/core/dialect.py

```diff
@@ -897,6 +897,14 @@ def select_from_values_for_batch_range(
         ]
 
     values_exp = exp.values(expressions, alias=alias, columns=columns_to_types)
+    if values:
```

Comment:
if we cast values, directly, do we need to also cast the select?

Reply:
are you referring to the projections? because we already cast them [earlier in this function](https://github.com/TobikoData/sqlmesh/pull/2856/files/6f9e7ee997868e3f3674ad8a506cb7841b6395a6#diff-32e7379e6d699794516fd766eee46a0a093be9bb23c3fdb4f8ea20117c2a44c3R884-R887)

Reply:
yea but now it's a double cast, cast in the values, and cast in the projection, should we just do one?

Reply:
Ah gotcha.. I guess we can try? I was being conservative because I'm not sure what the implications are in case we remove those casts, maybe we'll fail to infer the type of a projection down the line somewhere? On the other hand, we seem to have the `columns_to_types` mapping already, so perhaps it's fine.

Reply:
no big deal